test/cbmc/proofs/stringBuilderUInt32Decimal/stringBuilderUInt32Decimal_harness.c

```diff
@@ -0,0 +1,50 @@
+/* Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved. */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+/*
+ * Insert copyright notice
+ */
+
+/**
+ * @file stringBuilderUInt32Decimal_harness.c
+ * @brief Implements the proof harness for stringBuilderUInt32Decimal function.
+ */
+
+/*
+ * Insert project header files that
+ *   - include the declaration of the function
+ *   - include the types needed to declare function arguments
+ */
+/* Include files required for stringBuilderUInt32Decimal function. */
+#include "ota_mqtt_private.h"
+
+/* Max Length of the pBuffer defined in the ota_mqtt.c file. */
+#define U32_MAX_LEN    10U
+
+/* Declaration of the Mangled name function generated by CBMC for static functions. */
+size_t __CPROVER_file_local_ota_mqtt_c_stringBuilderUInt32Decimal( char * pBuffer,
+                                                                   size_t bufferSizeBytes,
+                                                                   uint32_t value );
+
+void stringBuilderUInt32Decimal_harness()
+{
+    /* The pBuffer is used as string in various functions but the maximum size of the. */
+    char * pBuffer;
+    size_t bufferSizebytes;
+    uint32_t value;
+
+    /* The bufferSizebytes is the size of the pBuffer. The pBuffer is statically initialized with
+     *  a size of U32_MAX_LEN + 1 in all the functions which call stringBuilderUInt32Decimal. Hence,
+     *  the size can never be below that. */
+    __CPROVER_assume( bufferSizebytes > U32_MAX_LEN );
```

Comment:
Is this assumption backwards? This seems like we are assuming that the size is always greater than the max.

Reply:
`U32_MAX_LEN` is not the maximum size of the `pBuffer` but it is the maximum number of digits of an unsigned long value. pBuffer should be atleast of that size to store all the digits in it and also the null character.

Reply:
I see, thank you for clarifying. In this case, do we want to test for the invalid inputs of a buffer size that's smaller? What if the user passes in a buffer of size U32_MAX_LEN?

Reply:
So the User does not define the buffer. It's defined as a local variable in the static function. So we don't have to worry about that issue

Reply:
Sounds good, thanks.